src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2252,6 +2280,13 @@ private bool UseUpdatedEscapeRulesForInvocation(Symbol symbol)
             {
                 foreach (var (_, argument, refKind) in escapeArguments)
                 {
+                    if (ShouldInferDeclarationExpressionValEscape(argument, out _))
+                    {
+                        // assume any declaration variable is a valid mixing destination,
+                        // since we are going to infer a val-escape which makes it valid.
```

Comment:
```suggestion
                        // assume any expression variable is a valid mixing destination,
                        // since we will infer a legal val-escape for it (if it doesn't already have a narrower one).
```